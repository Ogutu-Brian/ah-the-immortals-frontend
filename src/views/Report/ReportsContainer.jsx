import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Spinner from '../Article/Spinner';
import { getReports,deleteReport } from '../../redux/actions/postActions';

class ReportsContainer extends Component {
  componentDidMount() {
    const { getReports } = this.props;
    getReports();
  }
  onDeleteClick = (slug) => {const { history, deleteReport } = this.props;if (window.confirm('Delete This article?')) {deleteReport(slug);history.push('/');}}
  render() {
    const { post: { loading } } = this.props;
    const { post: { AllReports } } = this.props;
    return (
      <div>
        <br />
        <br />
        <h1 className="text-center">Reports</h1>
        <br />
        <table className="table table-hover">
          <thead className="table-primary">
            <tr>
              <th scope="col">#</th>
              <th scope="col">Article Title</th>
              <th scope="col">Reason</th>
              <th scope="col">Description</th>
              <th scope="col">Action</th>
            </tr>
          </thead>
          {
            (!AllReports['escalated articles'].length || loading) ? loading : (
              AllReports['escalated articles'].length ? AllReports['escalated articles'].map((item, index) => (

                <tbody key={`${index + 1}`}>
                  <tr>
                    <th scope="row">{index + 1}</th>
                    <td>{item.article.title}</td>
                    <td>{item.reason}</td>
                    <td>{item.description}</td>
                    <td>
                      <Link to="#null">
                        <i className="fa fa-trash fa-1x" onClick={() => this.onDeleteClick(item.article.slug)} aria-hidden="true"> Delete</i>
                          
                      </Link>
                    </td>
                  </tr>
                </tbody>

              )) : null
            )}
        </table>
        <br />
        <br />
        <br />
      </div>
    );
  }
}

ReportsContainer.propTypes = {
  getReports: PropTypes.func.isRequired,
  deleteReport:PropTypes.func.isRequired,
  post: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};
const mapStateToProps = state => ({
  post: state.post
});

export default connect(mapStateToProps, { getReports,deleteReport })(ReportsContainer);
