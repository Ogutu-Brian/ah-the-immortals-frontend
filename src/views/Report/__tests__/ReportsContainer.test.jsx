import React from 'react';
import { Provider } from 'react-redux';
import { mount,shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { createBrowserHistory } from 'history';
import { Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import ReportsContainer from '../ReportsContainer';

describe('Reports', () => {
  const testStore = configureMockStore([thunk]);
  const initialState = {
    post: {
      posts: [],
      post: {
      },
      pages: {
        
      },
      pagination: {},
      editpost: [],
      loading: false,
      Report:[],
      AllReports:{
        'escalated articles':[
          {
            article: {
              slug: 'new-one-3',
              title: 'New one',
              description: 'This is a new article',
              body: 'Of course it\'s a new article',
              created_at: '2019-06-20T10:26:57.864585Z',
              updated_at: '2019-06-20T10:26:57.864640Z'
            },
            reason: 'Plagiarism',
            description: 'test flex'
          }
        ]
      },
    }
  };
  const store = testStore(initialState);
  const history = createBrowserHistory();
  const wrapper = mount(
    <Router history={history}>
      <Provider store={store}>
        <ReportsContainer history={history} />
      </Provider>
    </Router>
  );
  it('passes', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
