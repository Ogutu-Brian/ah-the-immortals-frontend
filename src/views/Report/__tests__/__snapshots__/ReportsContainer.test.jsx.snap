// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports passes 1`] = `ReactWrapper {}`;
